.github/workflows/bluetooth.yaml

```diff
@@ -32,6 +24,10 @@ jobs:
       BSIM_COMPONENTS_PATH: /opt/bsim/components
       EDTT_PATH: ../tools/edtt
       bsim_bt_test_results_file: ./bsim_bt_out/bsim_results.xml
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
```

Comment:
If the `github.ref` is "main" for the `pull_request_target` events, that will indeed be the case.

Reply:
I suppose we could do something like `${{ github.workflow }}-${{ github.head_ref || github.head_ref }}`, but it needs more verification.

Reply:
or `${{ github.workflow }}-${{ github.head_ref || github.run_id }}`. Only `pull_request` and `pull_request_target` have `head_ref`. If it is not found (e.g. on push or cron) it will take `run_id` which is unique to the run https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-fallback-value-1
I guess the difference is that for non `pull_request(_...)` triggers such actions won't be killed and I assume this is what we want. So each push to the main will have the corresponding results, even when another happened before the action finished.

Reply:
For this particular case, it does not matter because it only triggers on the `pull_request_target` type events (i.e. no `push`).

If it were set to trigger on both `push` or `pull_request_target` events, `${{ github.workflow }}-${{ github.head_ref || github.ref }}` should do the job since it will allow only one job to execute per pull request (i.e. with `head_ref` set to the pull request branch, while also allowing only one job on the push (in which case `head_ref` is not defined and `ref` points to the pushed branch, e.g. main).

Reply:
`${{ github.workflow }}-${{ github.head_ref || github.run_id }}` is problematic because, in case of push to main for instance, it will allow two parallel jobs to execute on the main, which is not what we want.

Reply:
I have created a test workflow for this:
push run = https://github.com/stephanosio/gha-concurrency-test/actions/runs/1858385051
pull_request_target run = https://github.com/stephanosio/gha-concurrency-test/runs/5231316748?check_suite_focus=true

Reply:
@stephanosio So we have the same understanding of how the logic works, its good :) Do you mean my assumption is wrong? I thought that we want the results to be available for each push.  I.e. we have a history of commits in the main and related workflows' results. E.g. we might have a busy merging period. Does it mean we are fine without getting a workflow results for any of such pushes, until we end in a quit period without any push, where the workflow could have run till its end? Maybe not this particular action, but there are others that works not only on pull_requests, like running twister tests.